From b046131c0bd44ca3a11276194d07b85373cfd608 Mon Sep 17 00:00:00 2001
From: Frank Barchard <fbarchard@google.com>
Date: Mon, 22 Mar 2021 13:10:06 -0700
Subject: Replace MOV .4s with MOV .16b for GCC compatability

MOV Vy.4s, Vx.4s is not a valid instruction form (even though LLVM allows it).
It should be MOV Vy.16b, Vx.16b (.8b for 64-bit variants)

Bug: None
Change-Id: I3c3b42288a0ebc275962fa3adad707b351d00d4c
Reviewed-on: https://chromium-review.googlesource.com/c/libyuv/libyuv/+/2780155
Reviewed-by: Mirko Bonadei <mbonadei@chromium.org>
---
 source/scale_neon64.cc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/source/scale_neon64.cc b/source/scale_neon64.cc
index cde4ee39..8656fec7 100644
--- a/source/scale_neon64.cc
+++ b/source/scale_neon64.cc
@@ -791,8 +791,8 @@ void ScaleRowUp2_Bilinear_16_NEON(const uint16_t* src_ptr,
       "umlal       v4.4s, v1.4h, v31.4h          \n"  // 3*near+far (2, odd)
       "umlal       v5.4s, v0.4h, v31.4h          \n"  // 3*near+far (2, even)
 
-      "mov         v0.4s, v4.4s                  \n"
-      "mov         v1.4s, v5.4s                  \n"
+      "mov         v0.16b, v4.16b                \n"
+      "mov         v1.16b, v5.16b                \n"
       "mla         v4.4s, v2.4s, v30.4s          \n"  // 9 3 3 1 (1, odd)
       "mla         v5.4s, v3.4s, v30.4s          \n"  // 9 3 3 1 (1, even)
       "mla         v2.4s, v0.4s, v30.4s          \n"  // 9 3 3 1 (2, odd)
@@ -988,8 +988,8 @@ void ScaleUVRowUp2_Bilinear_16_NEON(const uint16_t* src_ptr,
       "umlal       v4.4s, v1.4h, v31.4h          \n"  // 3*near+far (2, odd)
       "umlal       v5.4s, v0.4h, v31.4h          \n"  // 3*near+far (2, even)
 
-      "mov         v0.4s, v4.4s                  \n"
-      "mov         v1.4s, v5.4s                  \n"
+      "mov         v0.16b, v4.16b                \n"
+      "mov         v1.16b, v5.16b                \n"
       "mla         v4.4s, v2.4s, v30.4s          \n"  // 9 3 3 1 (1, odd)
       "mla         v5.4s, v3.4s, v30.4s          \n"  // 9 3 3 1 (1, even)
       "mla         v2.4s, v0.4s, v30.4s          \n"  // 9 3 3 1 (2, odd)
-- 
2.25.1

